Upstream: yes

From c9aca1b53324bcd1b5a2de5c645813f80eccd016 Mon Sep 17 00:00:00 2001
From: Sami Kerola <kerolasa@iki.fi>
Date: Mon, 25 Mar 2019 19:59:36 +0000
Subject: [PATCH] tracepath: fix musl build, again

Move linux headers after standard headers, that should sort out regression.
Also add a comment that might help keeping includes in good order in future.

Regression-since: 4f0cfdb215c47e04f6b0f73306f8e902cb836b17
Reported-by: Will Dietz <github@wdtz.org>
Reference: https://github.com/iputils/iputils/issues/168
Signed-off-by: Sami Kerola <kerolasa@iki.fi>
---
 tracepath.c | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/tracepath.c b/tracepath.c
index a72fbcb..e17999a 100644
--- a/tracepath.c
+++ b/tracepath.c
@@ -12,10 +12,6 @@
 #include <arpa/inet.h>
 #include <errno.h>
 #include <limits.h>
-#include <linux/errqueue.h>
-#include <linux/icmp.h>
-#include <linux/icmpv6.h>
-#include <linux/types.h>
 #include <netdb.h>
 #include <netinet/in.h>
 #include <resolv.h>
@@ -27,6 +23,15 @@
 #include <sys/uio.h>
 #include <unistd.h>
 
+/*
+ * Keep linux/*.h includes after standard headers.
+ * https://github.com/iputils/iputils/issues/168
+ */
+#include <linux/errqueue.h>
+#include <linux/icmp.h>
+#include <linux/icmpv6.h>
+#include <linux/types.h>
+
 #include "iputils_common.h"
 
 #ifdef USE_IDN
-- 
2.21.0

